src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1090,6 +1120,126 @@ public void UpdateBaseline(long timestampDiff)
+                        {
+                            currentLocalTime = originalBaseline.AddMilliseconds(localTimestamp);
+                            if (isAppend)
+                            {
+                                splitSegment.Append(_context.Allocator, 0, localValues, localTag.AsSpan(), localStatus);
+                                continue;
+                            }
+
+                            var compareTags = localTag.ContentAsSpan().SequenceCompareTo(tagSlice);
+                            if (compareTags < 0) // local tag wins
+                            {
+                                splitSegment.Append(_context.Allocator, 0, localValues, localTag.AsSpan(), localStatus);
+                                continue;
+                            }
+
+                            if (compareTags == 0) // equal tags
+                            {
+                                if (FromReplication)
+                                {
+                                    var compareValues = localValues.SequenceCompareTo(newValues);
+                                    var isIncrement = tagSlice.StartsWith(TimedCounterPositivePrefixBuffer);
+
+                                    if (isIncrement) // increment entry --> take higher values
+                                    {
+                                        if (compareValues >= 0)
+                                            splitSegment.Append(_context.Allocator, 0, localValues, localTag.AsSpan(), localStatus);
+                                        else
+                                            splitSegment.Append(_context.Allocator, 0, newValues, tagSlice, status);
+                                    }
+                                    else // decrement entry --> take lower values
+                                    {
+                                        if (compareValues <= 0)
+                                            splitSegment.Append(_context.Allocator, 0, localValues, localTag.AsSpan(), localStatus);
+                                        else
+                                            splitSegment.Append(_context.Allocator, 0, newValues, tagSlice, status);
+                                    }
+
+                                    isAppend = true;
+                                    continue;
+                                }
+
+                                int length = Math.Min(localValues.Length, newValues.Length);
+                                for (int i = 0; i < length; i++)
+                                {
+                                    newValues[i] += localValues[i];
+                                }
+                            }
+
+                            isAppend = true;
+                            splitSegment.Append(_context.Allocator, 0, newValues, tagSlice, status);
+                        }
```

Comment:
I would prefer to avoid all this.
Instead `TrimSegmentToDate` should slice one segment into two parts at a given point and then retry the append.

Reply:
Done